src/main/java/sg/ncl/MainController.java

```diff
@@ -86,9 +86,12 @@
     @Inject
     private ObjectMapper objectMapper;
 
-    @Autowired
+    @Inject
     private ConnectionProperties properties;
 
+    @Inject
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use constructor injection for this field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3306)